offchain/README.md

```diff
@@ -1,8 +1,119 @@
+
+```shell
+git submodule update --init --recursive
+```
+
+The Cartesi Node depends on the following components:
+
+| Component | Version |
+|---|---|
+| Cartesi Machine Emulator ROM | [v0.16.0](https://github.com/cartesi/machine-emulator-rom/releases/tag/v0.16.0) |
+| Cartesi Machine Image Rootfs | [v0.17.0](https://github.com/cartesi/image-rootfs/releases/tag/v0.17.0) |
+| Cartesi Machine SDK | [v0.14.0](https://github.com/cartesi/machine-emulator/releases/tag/v0.14.0) |
+| Cartesi OpenAPI Interfaces | [v0.6.0](https://github.com/cartesi/openapi-interfaces/releases/tag/v0.6.0) |
+| Linux Kernel | [v5.15.63-ctsi-2](https://github.com/cartesi/linux/releases/tag/v5.15.63-ctsi-2) |
+
+### Running
+
+To run any of the inner projects, execute the command:
+
+```shell
+cargo run
+```
+
+Some of the inner projects may have additional run instructions.
+Refer to their own documentation for more details.
+
+#### Configuration
+
+It is possible to configure the behavior of any of the projects by passing CLI arguments and using environment variables.
+Execute the following command to check the available options for each project:
+
+```shell
+cargo run -- -h
+```
+
+##### Redis TLS Configuration
+
+To connect the Broker to a Redis server via TLS, the server's URL must use the `rediss://` scheme (with two "s"es).
 This is currently the only way to tell `Broker` to use a TLS connection.
+
+### Tests
+
+To run the tests available in any of the projects, execute the command:
+
+```shell
+cargo test
+```
+
+For integration tests, it is possible to see additional logs by setting the following variable:
+
+```shell
+export CARTESI_TEST_VERBOSE=1
+```
+
+## Architecture Overview
+
+The Cartesi Node is an event-based solution that may be depicted as follows:
+
+![Node architecture diagram](./docs/node-architecture.drawio.svg)
+
+The **Broker** is a Redis-based message broker that mediates the transferring of *Inputs*, *Outputs* and *Claims* between the Cartesi Node components.
+
+The [**State-fold Server**](./state-server/README.md) is the component responsible for tracking blockchain state changes and deriving *Inputs* from them to be processed.
+It's also responsible for submitting *Claims* to the blockchain whenever they are created by the Cartesi Node.
+
+The [**Dispatcher**](./dispatcher/README.md) is the component that messages *Inputs* via the **Broker** to be processed elsewhere and captures **Claims** to be submitted by the **State-fold Server**.
+
+The [**Advance Runner**](./advance-runner/README.md) is the one responsible for relaying *Inputs* to the **Server Manager** to be processed by the underlying DApp running in the embedded **Cartesi Machine** and for capturing wathever **Outputs** that may be generated as a result.
+
+The [**Indexer**](./indexer/README.md) consumes all *Inputs* and *Outputs* transferred by the **Broker** and store them in a **PostgreSQL** database for later querying via a **GraphQL Server** instance.
```

Comment:
Are we omitting Reports on purpose?

Reply:
We are generalizing as Outputs... Perhaps we could list all kinds, just in case

Reply:
Nothing against it, just wanted to understand if it was on purpose. Let's leave it like this 😉